pkg/controller/publishingstrategy/publishingstrategy_controller.go

```diff
@@ -100,6 +107,141 @@ func (r *ReconcilePublishingStrategy) Reconcile(request reconcile.Request) (reco
 		return reconcile.Result{}, err
 	}
 
+	// get a list of all ingress on the cluster
+	ingressControllerList := &operatorv1.IngressControllerList{}
+	listOptions := []client.ListOption{
+		client.InNamespace("openshift-ingress-operator"),
+	}
+	err = r.client.List(context.TODO(), ingressControllerList, listOptions...)
+	if err != nil {
+		log.Error(err, "Cannot get list of ingresscontroller")
+		return reconcile.Result{}, err
+	}
+
+	// create temp list of applicationIngress
+	ingressNotOnClusterList := instance.Spec.ApplicationIngress
+	// loop through every applicationingress in publishing strategy and every ingresscontroller in cluster
+	for _, publishingStrategyIngress := range instance.Spec.ApplicationIngress {
+		for _, ingressController := range ingressControllerList.Items {
+			if !isOnCluster(&publishingStrategyIngress, &ingressController) {
+				ingressNotOnClusterList = append(ingressNotOnClusterList, publishingStrategyIngress)
```

Comment:
We're starting with `ingressNotOnClusterList` containing all the ApplicationIngress from the PublishingStrategy, then later we re-append into the same slice ingresses that don't match. Isn't that creating a bunch of duplicate ingresses in that slice?

Reply:
I thought we would in this first nested loop actually remove from `ingressNotOnClusterList` slice the ingresses that matched, resulting in a slice containing only Ingresses that are NOT in cluster (as the slice name states)

Reply:
ah that is my mistake.`ingressNotOnClusterList` should be of type `[]cloudingressv1alpha1.ApplicationIngress` which is an empty slice. Then we append ones that don't match to the slice.

Reply:
/lgtm

Reply:
thanks for fixing that